src/Workspaces/Remote/ServiceHub/Host/TestUtils.cs

```diff
@@ -184,32 +184,38 @@ public static Task AppendAssetMapAsync(this Solution solution, Dictionary<Checks
         public static async Task AppendAssetMapAsync(
             this Solution solution, Dictionary<Checksum, object> map, ProjectId? projectId, CancellationToken cancellationToken)
         {
+            var callback = (Checksum checksum, object asset, CancellationToken cancellationToken) =>
```

Comment:
this is test code.  so we just bridge to the testing-map form by passing a callback that populates it.